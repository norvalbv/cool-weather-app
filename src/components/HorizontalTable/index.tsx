import React, { ReactElement } from 'react';
import classNames from 'utils/classNames';

export interface HorizontalTableProps {
  title: string;
  value: JSX.Element | JSX.Element[] | string | number | null;
  valueSymbol?: string;
  widths?: { title: string; value: string };
  requirePadding?: boolean;
}

const HorizontalTable = ({
  title,
  value,
  valueSymbol,
  widths = { title: 'w-3/5', value: 'w-2/5' },
  requirePadding = true,
}: HorizontalTableProps): ReactElement => {
  return (
    <div className="group flex w-40">
      <div
        className={classNames(
          'text-neutral-N400 select-none leading-[1.145]',
          {
            'py-2': requirePadding,
          },
          widths.title
        )}
      >
        {title}
      </div>
      <div
        className={classNames(
          'text-neutral-N600 text-right leading-[1.145]',
          {
            'py-2': requirePadding,
          },
          widths.value
        )}
      >
        {value}&nbsp;
        {valueSymbol}
      </div>
    </div>
  );
};

export default HorizontalTable;
