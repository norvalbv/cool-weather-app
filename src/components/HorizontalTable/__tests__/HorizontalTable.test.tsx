import React, { FC } from 'react';
import renderer from 'react-test-renderer';

import HorizontalTable, { HorizontalTableProps } from '..';

const HorizontalTableWithAllProps: FC<Required<HorizontalTableProps>> = HorizontalTable;

describe('<HorizontalTable />', () => {
  test('Should render correctly with mandatory props', () => {
    const tree = renderer.create(<HorizontalTable title="Test Title" value={3} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Should render correctly with all props', () => {
    const tree = renderer
      .create(
        <HorizontalTableWithAllProps
          title="Test Title"
          value={3}
          valueSymbol="#"
          widths={{
            title: '2/5',
            value: '3/5',
          }}
          requirePadding
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
