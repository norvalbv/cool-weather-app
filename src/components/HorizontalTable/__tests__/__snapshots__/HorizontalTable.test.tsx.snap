// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HorizontalTable /> Should render correctly with all props 1`] = `
<div
  className="group flex w-40"
>
  <div
    className="text-neutral-N400 select-none leading-[1.145] py-2 2/5"
  >
    Test Title
  </div>
  <div
    className="text-neutral-N600 text-right leading-[1.145] py-2 3/5"
  >
    3
     
    #
  </div>
</div>
`;

exports[`<HorizontalTable /> Should render correctly with mandatory props 1`] = `
<div
  className="group flex w-40"
>
  <div
    className="text-neutral-N400 select-none leading-[1.145] py-2 w-3/5"
  >
    Test Title
  </div>
  <div
    className="text-neutral-N600 text-right leading-[1.145] py-2 w-2/5"
  >
    3
     
  </div>
</div>
`;
